### ISSUE TITLE
About the thresh, f-score, and label data

### ISSUE THREAD

**Issue by Luffy03** [lehaifeng/DASNet#19]
hello, I run your code and CDD dataset under the instruction but get bad results. My 'Best Thresh' is always 1, and F-score has no promotion. Could you please tell me why? Thank you very much! Whats' more, I am curious about the label. The range should be set to [0,1] or [0,255]. The initial data is [0,255], without normalizing to [0,1] I get weird loss just like hundreds.
![截屏2020-07-10下午2 51 45](https://user-images.githubusercontent.com/52514670/87126875-8a1e3900-c2bf-11ea-94be-430ca5f5ef8f.png)

**Comment by wangshihuang** [comment]
当label是0-255时，loss负值，当label是0-1是，loss看起来正常，f1分数一直是nan，大神，这个要怎么修改
def eval_image_rewrite(gt_image, prob, cl_index):
    thresh = np.array(range(0, 256)) / 255.0
    cl_gt = gt_image[:, :] == cl_index
    valid_gt = gt_image[:, :] != 255

    FN, FP, posNum, negNum = evalExp(cl_gt, prob,
                                     thresh, validMap=None,
                                     validArea=valid_gt)
    return FN, FP, posNum, negNum

应该和这段代码有关，在utils/metric文件里面

**Comment by wangshihuang** [comment]
> 您好，我在指令下执行了您的程式码和CDD资料集，但结果不理想。我的“最佳阈值”始终为1，F分数没有提升。你能告诉我为什么吗？非常感谢你！更重要的是，我对该标签感到好奇。范围应设置为[0,1]或[0,255]。初始数据为[0,255]，而没有归一化为[0,1]，我得到了数百个奇怪的损失。
> ![截屏2020-07-10下午2 51 45](https://user-images.githubusercontent.com/52514670/87126875-8a1e3900-c2bf-11ea-94be-430ca5f5ef8f.png)

当label是0-255时，loss负值，当label是0-1是，loss看起来正常，f1分数一直是nan，大神，这个要怎么修改
def eval_image_rewrite(gt_image, prob, cl_index):
    thresh = np.array(range(0, 256)) / 255.0
    cl_gt = gt_image[:, :] == cl_index
    valid_gt = gt_image[:, :] != 255

    FN, FP, posNum, negNum = evalExp(cl_gt, prob,
                                     thresh, validMap=None,
                                     validArea=valid_gt)
    return FN, FP, posNum, negNum

应该和这段代码有关，在utils/metric文件里面

**Comment by lehaifeng** [comment]
HI Luffy. 
Would you please give us more about your setting on hyperparameters?

**Comment by wangshihuang** [comment]
中文，会吗



---Original---

**Comment by lehaifeng** [comment]
当然

**Comment by wangshihuang** [comment]
没做出来，修改后只能训练，用预训练模型测试全是一个颜色



---Original---

**Comment by wangshihuang** [comment]
你是作者，你没做出来吗



---Original---

**Comment by wangshihuang** [comment]
这个代码的论文是怎么做出来的？

**Comment by wangshihuang** [comment]
作者能不能分享一下测试代码

**Comment by wangshihuang** [comment]
> 作者能不能分享一下测试代码

用CDD_model_best.pth就能测试的，

**Comment by Luffy03** [comment]
> HI Luffy.
> Would you please give us more about your setting on hyperparameters?

非常感谢您的工作！通过训练我已经用CDDdata得到您在论文中展示的结果。还有几个问题想向您请教
请问loss.py中所有的函数都是您在论文中展示的结构吗，使用某些loss时没法得到效果
另外请问您是否在较大尺度的图像上测试过？

**Comment by zyp19** [comment]
@Luffy03，您好，请问您是否方便分享一下您的测试代码呢？十分感谢！！

**Comment by Luffy03** [comment]
> @Luffy03，您好，请问您是否方便分享一下您的测试代码呢？十分感谢！！

测试代码就是validate函数，我是在single_layer_similar_heatmap_visual里加几行代码将输出结果保存到路径

**Comment by dsagsdfgfdh** [comment]
label的值应该是0和1，我们测试了很多的loss函数，但是只有我们提出的函数得到了一个最优的结果，受限于实验设施，我们并没有在大尺寸上进行实验。

**Comment by Anysomeday** [comment]
@Luffy03 I run this project followed the instructions, but the F-score is always low as about 0.003 during the training process. Additionally, when I directly use the 'CDD_model_best.pth' downloaded from 'https://pan.baidu.com/s/1GFkBXvVKgD1IqLYYeioX_w#list/path=%2F', Missing key(s) occur in the FC layers and embedding layers. Moreover, I don't know how to test the model.
Have you gone through this project, would you please share your code with me? My email is 'xi_201401@163.com', thank you.

**Comment by Luffy03** [comment]
> @Luffy03 I run this project followed the instructions, but the F-score is always low as about 0.003 during the training process. Additionally, when I directly use the 'CDD_model_best.pth' downloaded from 'https://pan.baidu.com/s/1GFkBXvVKgD1IqLYYeioX_w#list/path=%2F', Missing key(s) occur in the FC layers and embedding layers. Moreover, I don't know how to test the model.
> Have you gone through this project, would you please share your code with me? My email is '[xi_201401@163.com](mailto:xi_201401@163.com)', thank you.

I have not changed the code, you may just follow the author's command.

**Comment by Anysomeday** [comment]
> > @Luffy03 I run this project followed the instructions, but the F-score is always low as about 0.003 during the training process. Additionally, when I directly use the 'CDD_model_best.pth' downloaded from 'https://pan.baidu.com/s/1GFkBXvVKgD1IqLYYeioX_w#list/path=%2F', Missing key(s) occur in the FC layers and embedding layers. Moreover, I don't know how to test the model.
> > Have you gone through this project, would you please share your code with me? My email is '[xi_201401@163.com](mailto:xi_201401@163.com)', thank you.
> 
> I have not changed the code, you may just follow the author's command.

I run the code following the author's instructions, but the loss is concussive, even negative, and the fscore is so low. As shown below:
![image](https://user-images.githubusercontent.com/28470436/92990257-72f10700-f50d-11ea-88b3-c610a383f27b.png)
Could you please tell me why is it happen and how to correct it?

**Comment by ClarissaDHL** [comment]
> 
> 
> hello, I run your code and CDD dataset under the instruction but get bad results. My 'Best Thresh' is always 1, and F-score has no promotion. Could you please tell me why? Thank you very much! Whats' more, I am curious about the label. The range should be set to [0,1] or [0,255]. The initial data is [0,255], without normalizing to [0,1] I get weird loss just like hundreds.
> ![截屏2020-07-10下午2 51 45](https://user-images.githubusercontent.com/52514670/87126875-8a1e3900-c2bf-11ea-94be-430ca5f5ef8f.png)

Hello, I encountered the same problem as you, how did you solve it at that time. And did u normalize the imgs?

**Comment by Luffy03** [comment]
> > hello, I run your code and CDD dataset under the instruction but get bad results. My 'Best Thresh' is always 1, and F-score has no promotion. Could you please tell me why? Thank you very much! Whats' more, I am curious about the label. The range should be set to [0,1] or [0,255]. The initial data is [0,255], without normalizing to [0,1] I get weird loss just like hundreds.
> > ![截屏2020-07-10下午2 51 45](https://user-images.githubusercontent.com/52514670/87126875-8a1e3900-c2bf-11ea-94be-430ca5f5ef8f.png)
> 
> Hello, I encountered the same problem as you, how did you solve it at that time. And did u normalize the imgs?

no i don't use normalize

**Comment by ClarissaDHL** [comment]
thank u for ur reply. I have already solved the problem. There is no need to normalize the input imgs but need to normalize the label from [0,255] to [0,1].





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "lehaifeng/DASNet"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年9月17日(星期四) 上午10:19
收件人:&nbsp;"lehaifeng/DASNet"<DASNet@noreply.github.com&gt;;
抄送:&nbsp;"185221369"<185221369@qq.com&gt;;"Comment"<comment@noreply.github.com&gt;;
主题:&nbsp;Re: [lehaifeng/DASNet] About the thresh, f-score, and label data (#19)





   
hello, I run your code and CDD dataset under the instruction but get bad results. My 'Best Thresh' is always 1, and F-score has no promotion. Could you please tell me why? Thank you very much! Whats' more, I am curious about the label. The range should be set to [0,1] or [0,255]. The initial data is [0,255], without normalizing to [0,1] I get weird loss just like hundreds.
 
  
Hello, I encountered the same problem as you, how did you solve it at that time. And did u normalize the imgs?
  
no i don't use normalize
